src/App.jsx

```diff
@@ -14,45 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabetically';
+const SORT_BY_LENGTH = 'length';
+
+const getPreparedGoods = (goods, { sortedField, isReverse }) => {
+  const preparedGoods = [...goods];
+
+  if (sortedField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortedField) {
+        case SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortedField, setSortedField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const [showReset, setShowReset] = useState(false);
```

Comment:
Instead using separate state for reset button I recommend relies on `sortedField` and `isReverse` values.

It will looks like this:
```
showReverse = sortedField || isReverse;
```